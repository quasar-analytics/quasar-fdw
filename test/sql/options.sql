/* CREATE SERVER options checks */
CREATE SERVER o_quasar0 FOREIGN DATA WRAPPER quasar_fdw ;
/* wrong options are illegal */
CREATE SERVER o_quasar1 FOREIGN DATA WRAPPER quasar_fdw OPTIONS (wrong 'foo');
/* repeated options is illegal */
CREATE SERVER o_quasar2 FOREIGN DATA WRAPPER quasar_fdw OPTIONS (server 'http://localhost:8080', path '/local/quasar', server 'http://localhost:8080');
/* timeout_ms 0 is illegal */
CREATE SERVER o_quasar3 FOREIGN DATA WRAPPER quasar_fdw OPTIONS (server 'http://localhost:8080', path '/local/quasar', timeout_ms '0');
/* CREATE TABLE options checks */
CREATE FOREIGN TABLE o_ft0(id integer) SERVER o_quasar;
/* wrong options are illegal */
CREATE FOREIGN TABLE o_ft1(id integer) SERVER o_quasar OPTIONS (wrong 'foo');
/* repeated options are illegal */
CREATE FOREIGN TABLE o_ft2(id integer) SERVER o_quasar OPTIONS (table 'bar', table 'baz');
/* Attribute Options */
/* wrong options are illegal */
CREATE FOREIGN TABLE o_bad_attr_opt(city varchar OPTIONS (none 'yoyo'))
       SERVER o_quasar OPTIONS (table 'zips');
