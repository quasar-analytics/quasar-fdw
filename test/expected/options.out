/* CREATE SERVER options checks */
CREATE SERVER o_quasar0 FOREIGN DATA WRAPPER quasar_fdw ;
CREATE SERVER o_quasar1 FOREIGN DATA WRAPPER quasar_fdw OPTIONS (wrong 'foo');
ERROR:  invalid option "wrong"
HINT:  Valid options in this context are: server, path
CREATE SERVER o_quasar2 FOREIGN DATA WRAPPER quasar_fdw OPTIONS (server 'http://localhost:8080', path '/local/quasar', server 'http://localhost:8080');
ERROR:  option "server" provided more than once
CREATE SERVER o_quasar FOREIGN DATA WRAPPER quasar_fdw OPTIONS (server 'http://localhost:8080', path '/local/quasar');
/* CREATE TABLE options checks */
CREATE FOREIGN TABLE o_ft0(id integer) SERVER o_quasar;
CREATE FOREIGN TABLE o_ft1(id integer) SERVER o_quasar OPTIONS (wrong 'foo');
ERROR:  invalid option "wrong"
HINT:  Valid options in this context are: table
CREATE FOREIGN TABLE o_ft2(id integer) SERVER o_quasar OPTIONS (table 'bar');
CREATE FOREIGN TABLE o_ft3(id integer) SERVER o_quasar OPTIONS (table 'bar', table 'baz');
ERROR:  option "table" provided more than once
CREATE FOREIGN TABLE o_zips(city varchar, pop integer, state char(2))
       SERVER o_quasar OPTIONS (table 'zips');
/* Attribute Options */
CREATE FOREIGN TABLE o_bad_attr_opt(city varchar OPTIONS (none 'yoyo'))
       SERVER o_quasar OPTIONS (table 'zips');
ERROR:  invalid option "none"
HINT:  Valid options in this context are: map
CREATE FOREIGN TABLE o_nested(a varchar OPTIONS (map 'topObj.midObj.botObj.a'),
                            b varchar OPTIONS (map 'topObj.midObj.botObj.b'),
                            c varchar OPTIONS (map 'topObj.midObj.botObj.c'))
       SERVER o_quasar
       OPTIONS (table 'nested');
